packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -18,20 +18,464 @@
+            info={},
+            ledger_id="ethereum",
+            signing_payload=tx
+        )
+
+        return tx_message
+
+    def _get_mint_batch_tx(
+        self, deployer_address, recipient_address, batch_mint_quantities, ledger_api,
+    ) -> str:
+        """Mint a batch of items."""
+        # mint batch
+        nonce = ledger_api.api.eth.getTransactionCount(deployer_address)
+        tx = self.instance.functions.mintBatch(
+            recipient_address, self.item_ids, batch_mint_quantities
+        ).buildTransaction(
+            {
+                "chainId": 3,
+                "gas": 300000,
+                "gasPrice": ledger_api.api.toWei("50", "gwei"),
+                "nonce": nonce,
+            }
+        )
+
+        return tx
+
+    def _get_trade_tx(self, terms, signature, ledger_api: LedgerApi) -> str:
+        """
+        Create a trade tx.
+
+        :params terms: The class (can be Dict[str, Any]) that contains the details for the transaction.
+        :params signature: The signed terms from the counterparty.
+        """
+        data = b"hello"
+        nonce = ledger_api.api.eth.getTransactionCount(terms.from_address)
+        tx = self.instance.functions.trade(
+            terms.from_address,
+            terms.to_address,
+            terms.item_id,
+            terms.from_supply,
+            terms.to_supply,
+            terms.value_eth_wei,
+            terms.trade_nonce,
+            signature,
+            data,
+        ).buildTransaction(
+            {
+                "chainId": 3,
+                "gas": 2818111,
+                "from": terms.from_address,
+                "value": terms.value_eth_wei,
+                "gasPrice": ledger_api.api.toWei("50", "gwei"),
+                "nonce": nonce,
+            }
+        )
+
+        return tx
+
+    def _get_trade_batch_tx(self, terms, signature, ledger_api: LedgerApi) -> str:
+        """
+        Create a batch trade tx.
+
+        :params terms: The class (can be Dict[str, Any]) that contains the details for the transaction.
+        :params signature: The signed terms from the counterparty.
+        """
+        data = b"hello"
+        nonce = ledger_api.api.eth.getTransactionCount(terms.from_address)
+        tx = self.instance.functions.tradeBatch(
+            terms.from_address,
+            terms.to_address,
+            terms.item_ids,
+            terms.from_supplies,
+            terms.to_supplies,
+            terms.value_eth_wei,
+            terms.trade_nonce,
+            signature,
+            data,
+        ).buildTransaction(
+            {
+                "chainId": 3,
+                "gas": 2818111,
+                "from": terms.from_address,
+                "value": terms.value_eth_wei,
+                "gasPrice": ledger_api.api.toWei("50", "gwei"),
+                "nonce": nonce,
+            }
+        )
+        return tx
+
+    def _get_balance(self, contract, from_address: Address, item_id: int):
```

Comment:
Introduce private function _get_balance() in case we want to do checks later on. ( The original version has assertions based on the balances but this must not be a case here.)